### PR TITLE
Make Watchdog Signal Configurable

### LINKED ISSUE #8658: RFE: add WatchdogSignal knob to control what is sent when the watchdog expires
Right now when a watchdog expires systemd sends a SIGABRT to the process, per https://www.freedesktop.org/software/systemd/man/systemd.service.html#WatchdogSec=. It would be useful to make this configurable (e.g. with a new `WatchdogSignal` setting), as e.g. it's not always desirable to cause a coredump (which is one of the side effects of SIGABRT), and a SIGKILL could be preferable in those cases.

### PR DESCRIPTION
**Pull request by anitazha**
Allows configuring watchdog signal to be SIGABRT, SIGKILL, or SIGTERM
(with a default of SIGABRT). This allows an alternative to SIGABRT
when coredumps are not desirable. The options are limited to signals
that translate to service stop state.

Closes #8658

### CODE REVIEW

File: src/core/load-fragment-gperf.gperf.m4 [hunk 1]
```diff
@@ -298,6 +298,7 @@ Service.TimeoutStartSec,         config_parse_service_timeout,       0,
 Service.TimeoutStopSec,          config_parse_sec_fix_0,             0,                             offsetof(Service, timeout_stop_usec)
 Service.RuntimeMaxSec,           config_parse_sec,                   0,                             offsetof(Service, runtime_max_usec)
 Service.WatchdogSec,             config_parse_sec,                   0,                             offsetof(Service, watchdog_usec)
+Service.WatchdogSignal,          config_parse_signal,                0,                             offsetof(Service, watchdog_signal)
```
Comment (yuwata) [inline src/core/load-fragment-gperf.gperf.m4]: `config_parse_signal()` can accept many signals. I think it is better to introduce more specific parser, say `config_parse_watchdog_signal()` and drop the change in `service_verify()`.

Comment (anitazha) [reply src/core/load-fragment-gperf.gperf.m4]: Makes sense. I'll update with your suggestion.

File: src/core/dbus-service.c [hunk 2]
```diff
@@ -306,6 +308,9 @@ static int bus_service_set_transient_property(
         if (streq(name, "WatchdogUSec"))
                 return bus_set_transient_usec(u, name, &s->watchdog_usec, message, flags, error);
 
+        if (streq(name, "WatchdogSignal"))
+                return bus_set_transient_int(u, name, &s->watchdog_signal, message, flags, error);
```
Comment (yuwata) [inline src/core/dbus-service.c]: OK. Now, the value `watchdog_signal` is not checked in `service_verify()`. So, please also check the input here. I guess `BUS_DEFINE_SET_TRANSIENT_IS_VALID()` macro can be used to define a function suitable here.

**Comment by poettering** [comment]
Hmm, what's the rationale for only allowing those three signals? if we make this configurable we might as well liberize this properly, and allow any other signal too...

**Comment by poettering** [comment]
(or to say this differently: i think we should just document that the signal specified should result in the service to shut down, but not insist on the exact signal to use. I think it's enough to push the people in the right direction, but not insist on it here)

**Comment by poettering** [comment]
patch looks good in general, just would prefer the aforementioned generalization (which is a simplification i guess)

**Comment by anitazha** [comment]
Then it would be easier to set `WatchdogSignal=` to any signal

**Comment by anitazha** [comment]
Or maybe not have `WatchdogSignal=` at all and instead make a knob to use `KillSignal=` instead of `SIGABRT` for watchdog. I guess I'm not entirely clear how the termination sequence should go with the addition of `WatchdogSignal=`.

**Comment by davide125** [comment]
Thanks again for working on this @anitazha! IMO this should be a separate knob from `KillSignal`, as one might want to use a different signal to trigger some watchdog-specific logic. I don't have a strong opinion about which signals to allow here, but I can see stuff like `SIGUSR1` being useful in some cases.

**Pull request merged**